File: 1749-maximum-absolute-sum-of-any-subarray/main.py
class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        
        max_sum = min_sum = 0
        max_ending = min_ending = 0

        for num in nums:
            
            max_ending = max(max_ending + num, num)      
            min_ending = min(min_ending + num, num)
                  
            max_sum = max(max_sum, max_ending)
            min_sum = min(min_sum, min_ending)

        return max(max_sum, abs(min_sum))

File: 1749-maximum-absolute-sum-of-any-subarray/main.cpp
class Solution {
public:
    int maxAbsoluteSum(vector<int>& nums) {
        int maxSum = 0, minSum = 0, maxEnding = 0, minEnding = 0;

        for (const auto& num: nums) {
            maxEnding = std::max(maxEnding + num, num);
            minEnding = std::min(minEnding + num, num);

            maxSum = std::max(maxSum, maxEnding);
            minSum = std::min(minSum, minEnding);
        }

        return std::max(maxSum, std::abs(minSum));
    }
};
